compiler/rustc_codegen_ssa/src/mir/place.rs

```diff
@@ -188,7 +164,7 @@ impl<'a, 'tcx, V: CodegenObject> PlaceRef<'tcx, V> {
         debug!("struct_field_ptr: DST field offset: {:?}", offset);
 
         // Adjust pointer.
-        let ptr = bx.gep(bx.cx().type_i8(), self.llval, &[offset]);
+        let ptr = bx.ptradd(self.llval, offset);
```

Comment:
This can probably be `inbounds` as well, I don't know why DSTs would be different. (I certainly hope we don't allow DST fields to go out of bounds...) Maybe I'll open another PR for this.